onnxruntime/core/framework/finalize_session_state.cc

```diff
@@ -99,7 +99,11 @@ Status FinalizeSessionState(SessionState& session_state,
   session_state.CleanInitializedTensorsFromGraph();
 
   ORT_RETURN_IF_ERROR(session_state.CreateKernels(kernel_registry_manager));
-  if (session_options.use_prepacking) {
+
+  bool disable_prepacking = HasSessionConfigEntry(session_options, kDisablePrePacking) &&
+                            session_options.session_configurations.at(kDisablePrePacking) == "1";
+
```

Comment:
A GetValueOrDefault may be useful here (default is provided as arg, return default if key not present)